ftdc/ftdc.go

```diff
@@ -106,9 +107,17 @@ type FTDC struct {
 	//
 	// When debug is true, the `outputWriter` will "tee" data to both the `currOutputFile` and
 	// `inmemBuffer`. Otherwise `outputWriter` will just refer to the `currOutputFile`.
-	debug          bool
-	outputWriter   io.Writer
-	currOutputFile *os.File
+	debug        bool
+	outputWriter io.Writer
+	// bytesWrittenCounter will count bytes that are written to the `outputWriter`. We use an
+	// `io.Writer` implementor for this, as opposed to just counting by hand, primarily as a
+	// convenience for working with `json.NewEncoder(writer).Encode(...)`. This counter is folded
+	// into the above `outputWriter`.
+	bytesWrittenCounter countingWriter
+	currOutputFile      *os.File
+	maxFileSize         int64
+	// ftdcDir controls where FTDC data files will be written.
+	ftdcDir string
```

Comment:
Where will this be in production?

Reply:
I figure we'll find a spot in `~/.viam/` to put it.

Unspecified for this PR. Will need to be agreed upon when making FTDC on by default